uber/site_sections/dept_checklist.py

```diff
@@ -3,77 +3,135 @@
+                attendee=attendee, department=department, slug=slug)
 
         if comments is not None:
-            check_csrf(csrf_token)  # since this form doesn't use our normal utility methods, we need to do this manually
+            # since this form doesn't use our normal utility methods, we need
+            # to check the csrf_token manually
+            check_csrf(csrf_token)
             item.comments = comments
-            session.add(item)
-            raise HTTPRedirect('index?message={}', conf.name + ' checklist data uploaded')
+            message = check(item)
+            if not message:
+                session.add(item)
+                message = conf.name + ' checklist data uploaded'
+            raise HTTPRedirect(
+                'index?department_id={}&message={}', department_id, message)
 
         return {
             'item': item,
-            'conf': conf
+            'conf': conf,
+            'department': department
         }
 
-    def overview(self, session, message=''):
+    def overview(self, session, filtered=False, message=''):
         checklist = list(DeptChecklistConf.instances.values())
+        attendee = session.admin_attendee()
+
+        dept_filter = [Department.members_who_can_admin_checklist.any(
+            Attendee.id == attendee.id)] if filtered else []
+
+        departments = session.query(Department).filter(*dept_filter) \
+            .options(
+                subqueryload(Department.members_who_can_admin_checklist),
+                subqueryload(Department.dept_checklist_items)) \
+            .order_by(Department.name)
+
         overview = []
-        for dept, dept_name in c.JOB_LOCATION_OPTS:
-            dept_heads = []
-            for attendee in session.single_dept_heads(dept):
-                statuses = []
-                for item in checklist:
-                    if item.completed(attendee):
-                        statuses.append({'done': True})
-                    elif days_before(7, item.deadline)():
-                        statuses.append({'approaching': True})
-                    elif item.deadline < datetime.now(UTC):
-                        statuses.append({'missed': True})
-                    else:
-                        statuses.append({})
-                    statuses[-1]['name'] = item.name
-                dept_heads.append([attendee, statuses])
-            overview.append([dept, dept_name, dept_heads])
+        for dept in departments:
+            is_checklist_admin = attendee.is_checklist_admin_of(dept)
+            can_admin_checklist = attendee.can_admin_checklist_for(dept)
+            statuses = []
+            for item in checklist:
+                status = {'conf': item, 'name': item.name}
+                checklist_item = dept.checklist_item_for_slug(item.slug)
+                if checklist_item:
+                    status['done'] = True
+                    status['completed_by'] = checklist_item.attendee.full_name
```

Comment:
As in the checklist_item_for_slug calculation, I think this will lead to some mildly strange behavior if multiple department heads complete the checklist. It's not a big deal though.

Reply:
Ok, I'll revisit this. I can't remember if I considered the case where multiple department heads completed the same item.

Reply:
The only weird behavior right now is that it might list just one attendee in the "completed by" section, right? You can totally put this off if needed, this is a huge PR already. 😅